marlowe/src/Language/Marlowe/Analysis/Safety/Types.hs

```diff
@@ -91,7 +109,254 @@ data SafetyError =
+      ]
+  toJSON (DuplicateAccount accountId token) =
+    object
+      [ "error" .= ("DuplicateAccount" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this account with this token." :: String)
+      , "account-id" .= accountId
+      , "token" .= token
+      , "fatal" .= True
+      ]
+  toJSON (DuplicateChoice choiceId) =
+    object
+      [ "error" .= ("DuplicateChoice" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this choice." :: String)
+      , "choice-id" .= choiceId
+      , "fatal" .= True
+      ]
+  toJSON (DuplicateBoundValue valueId) =
+    object
+      [ "error" .= ("DuplicateBoundValue" :: String)
+      , "detail" .= ("In the initial state of the contract, there are duplicate entries for this bound value." :: String)
+      , "value-id" .= valueId
+      , "fatal" .= True
+      ]
+  toJSON (MaximumValueMayExceedProtocol natural) =
+    object
+      [ "error" .= ("MaximumValueMayExceedProtocol" :: String)
+      , "detail" .= ("At some point in during its executation, the contract may hold more native tokens than permitted by the ledger rules." :: String)
+      , "bytes" .= natural
+      , "fatal" .= False
+      ]
+  toJSON (TransactionSizeMayExceedProtocol transaction natural) =
+    object
+      [ "error" .= ("TransactionSizeMayExceedProtocol" :: String)
+      , "detail" .= ("This transaction's size may exceed the limit permitted by the ledger rules." :: String)
+      , "transaction" .= transaction
+      , "bytes" .= natural
+      , "fatal" .= False
+      ]
+  toJSON (TransactionCostMayExceedProtocol transaction exBudget) =
+    object
+      [ "error" .= ("TransactionCostMayExceedProtocol" :: String)
+      , "detail" .= ("This transaction's Plutus execution cost may exceed the limit permitted by the ledger rules." :: String)
+      , "transaction" .= transaction
+      , "cost" .= exBudget
+      , "fatal" .= False
+      ]
+  toJSON (TransactionValidationError transaction message) =
+    object
+      [ "error" .= ("TransactionValidationError" :: String)
+      , "detail" .= ("This transaction fails to validate on the ledger." :: String)
+      , "transaction" .= transaction
+      , "message" .= message
+      , "fatal" .= True
+      ]
+  toJSON (TransactionWarning transaction warning) =
+    object
+      [ "error" .= ("TransactionWarning" :: String)
+      , "detail" .= ("A Marlowe semantics warning is reported for this transaction." :: String)
+      , "transaction" .= transaction
+      , "warning" .= warning
+      , "fatal" .= False
+      ]
+  toJSON (MissingContinuation (DatumHash datumHash)) =
+    object
+      [ "error" .= ("MissingContinuation" :: String)
+      , "detail" .= ("The contract is missing a continuation that is not present in its map of continuations." :: String)
+      , "hash" .= toJSON (EncodeBase16 $ fromBuiltin datumHash)
+      , "fatal" .= False
```

Comment:
Should this be fatal or non-fatal? One might choose to not supply the full contract for analysis.